lib/bar/BottomBar.js

```diff
@@ -111,19 +112,27 @@ export default class BottomBar extends React.Component {
   }
 
   _renderActionSheet() {
-    const { displayActionButton, onAction } = this.props;
+    const { customButton, displayActionButton, onAction } = this.props;
+
+    let components = [];
+
+    if (customButton) components.push(customButton);
 
     if (displayActionButton) {
-      return (
-        <TouchableOpacity style={[styles.button, { flex: 0 }]} onPress={onAction}>
-          <Image
-            source={require('../../Assets/share-button.png')}
-            style={styles.buttonImage}
-          />
+      components.push(
+        <TouchableOpacity style={styles.button} onPress={onAction}>
+          <Image source={require('../../Assets/share-button.png')} />
         </TouchableOpacity>
       );
     }
-    return null;
+    
+    return (
+      <View style={styles.rightButtonContainer}>
+        {components.map((component, index) => (
+          <View key={index}>{component}</View>
```

Comment:
Is wrapping component with  `View` necessary here? Wouldn't this just work?

``` 
{components.map(component =>component}
```

And you may move the key part to where the component is defined?